src/main/java/io/kestra/plugin/openai/AbstractTask.java

```diff
@@ -20,9 +31,34 @@ public abstract class AbstractTask extends Task implements OpenAiInterface{
 
     protected String user;
 
+    @Schema(
+        title = "The timeout value to send to the OpenAI API."
+    )
+    @PluginProperty
+    private Integer apiTimeout;
+
     protected OpenAiService client(RunContext runContext) throws IllegalVariableEvaluationException {
         String apiKey = runContext.render(this.apiKey);
 
-        return new OpenAiService(apiKey);
+        return createOpenAiService(apiKey, Duration.ofSeconds(apiTimeout));
+    }
+
+    private static OpenAiService createOpenAiService(String token, final Duration timeout) {
```

Comment:
`com.theokanning.openai` assumes that you want to use the result of OpenAI function calls to call Java methods; it doesn't give you the option of getting the JSON response from OpenAI unless you customize the creation of the `OpenAiApi` class. This required the creation both a mixin class and a class that extends the `com.theokanning.openai` library's `ChatFunction`. 

To keep things self-contained, I added them as inner classes but I'm happy to move them into separate files if you're prefer that. 

I was hoping there would be a simpler way to get the library to provide direct access to the function call data OpenAI returns, but this was the only working solution I found.